src/encode/json.rs

```diff
@@ -162,6 +171,34 @@ impl Deserialize for JsonEncoderDeserializer {
         Ok(Box::<JsonEncoder>::default())
     }
 }
+#[cfg(feature = "log_kv")]
+mod kv {
+    use log::kv::VisitSource;
+    use std::collections::BTreeMap;
+
+    pub(crate) type Map = BTreeMap<String, String>;
+
+    pub(crate) fn get_attributes(source: &dyn log::kv::Source) -> anyhow::Result<Map> {
+        struct Visitor {
+            inner: Map,
+        }
```

Comment:
Please move this struct defn out of the fn. Additionally, for performance lets add a `with_capacity` method that allows the user to do:
```
let visitor = Visitor::with_capacity(source.count());
```

Optionally visitor might be better as a tuple variant: `struct Visitor(Map)`

Reply:
The underlying [BTreeMap](https://doc.rust-lang.org/stable/std/collections/struct.BTreeMap.html) does not have a `with_capacity` method, so I'm not sure what a `with_capacity` would look like for Visitor.

Reply:
Oh, yep you're right. Force of habit to always want to with_capacity stuff, apologies for the error. While we're on the subject, what is the reasoning to use a BTree instead of a HashMap or Vec<(String, String)> that allows for preallocation?

Reply:
I initially just thought it would be a more minimal memory footprint, with little overhead as my understanding is that log attributes are often not too numerous. However this is all basically moot now, I've switched to direct serialization from the log::kv::Source object with dynamic dispatch, following your recommendation.